packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -194,10 +172,36 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
         this.send(perfEvent);
     }
 
-    protected prepareEvent(event: ITelemetryBaseEvent): ITelemetryBaseEvent {
+    protected redactTaggedUserData(event: ITelemetryBaseEvent, propKey: string) {
+        const prop = event[propKey];
+        const { value, tag } = (typeof prop === "object")
```

Comment:
```suggestion
        const { value, tag } = (isTaggedTelemetryPropertyValue(prop))
```

To @vladsud's point, this is safer (but slower 🤷‍♂️)